iam_assumegroup/main.tf

```diff
@@ -0,0 +1,40 @@
+# -- Variables --
+
+variable "master_account_id" {
+  description = "AWS account ID for the master account."
+}
+
+variable "group_role_map" {
+  description = "Roles map for IAM groups, along with account types per role to grant access to."
+  type = map(list(map(list(string))))
+}
+
+locals {
+  group_account_map = [
+    for line in flatten([
+      for item, access in
+      {
+        for group, perms in var.group_role_map : group => flatten([
+          for perm in perms: flatten([
+            for pair in setproduct(keys(perm), flatten([values(perm)])): join("", [pair[1], "Assume", pair[0]])
+          ])
+        ])
+      }: formatlist("%s %s", item, access)
+    ]): split(" ", line)
+  ]
+}
+
+# -- Resources --
+
+resource "aws_iam_group" "iam_group" {
+  for_each = var.group_role_map
+
+  name = each.key
+}
+
+resource "aws_iam_group_policy_attachment" "iam_group_policies" {
```

Comment:
Suggested change to make:
`module.main.module.iam_groups.aws_iam_group_policy_attachment.iam_group_policies[12]` instead look like:
`module.main.module.iam_groups.aws_iam_group_policy_attachment.iam_group_policies["devops_SandboxAssumeReadOnly"]` in order to make future changes to permissions easier to understand in plans and avoid list churn.

~~~
diff --git a/iam_assumegroup/main.tf b/iam_assumegroup/main.tf
index 4ec0624..ebb57c7 100644
--- a/iam_assumegroup/main.tf
+++ b/iam_assumegroup/main.tf
@@ -10,7 +10,7 @@ variable "group_role_map" {
 }

 locals {
-  group_account_map = [
+  group_account_map = {
     for line in flatten([
       for item, access in
       {
@@ -20,8 +20,8 @@ locals {
           ])
         ])
       }: formatlist("%s %s", item, access)
-    ]): split(" ", line)
-  ]
+    ]): join("_", split(" ", line)) => split(" ", line)
+  }
 }

 # -- Resources --
@@ -33,8 +33,8 @@ resource "aws_iam_group" "iam_group" {
 }

 resource "aws_iam_group_policy_attachment" "iam_group_policies" {
-  count = length(local.group_account_map)
+  for_each = local.group_account_map

-  group = element(local.group_account_map[count.index],0)
-  policy_arn = "arn:aws:iam::${var.master_account_id}:policy/${element(local.group_account_map[count.index],1)}"
-}
\ No newline at end of file
+  group = each.value[0]
+  policy_arn = "arn:aws:iam::${var.master_account_id}:policy/${each.value[1]}"
+}
~~~